enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java

```diff
@@ -83,49 +84,26 @@ public void lockCleanupOnModeSwitch() throws Throwable
     @Test
     public void aPendingMemberShouldBeAbleToServeReads() throws Throwable
     {
+        // given
         createNodeOnMaster( testLabel, cluster.getMaster() );
         cluster.sync();
+
         HighlyAvailableGraphDatabase slave = cluster.getAnySlave();
+        cluster.fail( slave, ClusterManager.NetworkFlag.values() );
+        cluster.await( instanceEvicted( slave ) );
 
-        AtomicBoolean run = new AtomicBoolean( true );
-        AtomicReference<Throwable> ref = new AtomicReference<>(  );
+        assertEquals( PENDING, slave.getInstanceState() );
```

Comment:
Are we sure it should go to PENDING before cluster considers it evicted? Perhaps assertEventually(..) with timing based on some ha/cluster timing setting?